src/exo/new_eff.py

```diff
@@ -1528,3 +1532,36 @@ def Check_Bounds(proc, alloc_stmt, block):
     if not is_ok:
         raise SchedulingError(
             f"The buffer {alloc_stmt.name} is accessed out-of-bounds")
+
+def Check_IsDeadAfter(proc, stmts, bufname, ndim):
```

Comment:
Does this function basically replace the dependency analysis for free variables checks?

Reply:
See the following test: `test_data_reuse_loop_fail()`